### PR TITLE
aiogram3.x support

### PR DESCRIPTION
**Pull request by darksidecat**

### CODE REVIEW

File: aiogram_dialog/context/intent_filter.py [hunk 1]
```diff
@@ -45,9 +45,34 @@ def __init__(self, storage: BaseStorage, state_groups: Dict[str, Type[StatesGrou
...
+            if isinstance(event, DialogUpdateEvent):
+                await self.on_pre_process_aiogd_update(event, data)
+            elif event.message:
+                await self.on_pre_process_message(event.message, data)
+            elif event.chat_member:
+                await self.on_pre_process_message(event.message, data)
+            elif event.callback_query:
+                await self.on_pre_process_callback_query(event.callback_query, data)
+        except CancelHandler:
+            return
```
Comment (Tishka17) [inline aiogram_dialog/context/intent_filter.py:65]: Are you sure `CancelHandler` should be ignored?

File: aiogram_dialog/context/intent_filter.py [hunk 2]
```diff
@@ -45,9 +45,34 @@ def __init__(self, storage: BaseStorage, state_groups: Dict[str, Type[StatesGrou
...
+            event: Update,
+            data: Dict[str, Any],
+    ) -> Any:
+
+        try:
+            if isinstance(event, DialogUpdateEvent):
+                await self.on_pre_process_aiogd_update(event, data)
+            elif event.message:
+                await self.on_pre_process_message(event.message, data)
+            elif event.chat_member:
```
Comment (Tishka17) [inline aiogram_dialog/context/intent_filter.py:60]: my_chat_member

File: aiogram_dialog/context/intent_filter.py [hunk 3]
```diff
@@ -117,16 +144,12 @@ async def on_pre_process_callback_query(self, event: CallbackQuery, data: dict):
             if stack.last_intent_id() != intent_id:
                 logger.warning(f"Outdated intent id ({intent_id}) for stack ({stack.id})")
                 raise CancelHandler()
+            event.Config.allow_mutation = True  # ToDo
```
Comment (Tishka17) [inline aiogram_dialog/context/intent_filter.py:147]: huh

File: aiogram_dialog/manager/manager.py [hunk 4]
```diff
@@ -50,9 +51,16 @@ def storage(self) -> StorageProxy:
...
+        chat = self.data['event_chat']
+        bot = self.data['bot']
+
+        message = None
+        if self.current_stack().last_message_id:
+            message = Message(chat=chat,
+                              message_id=self.current_stack().last_message_id,
+                              date=datetime.now()  # ToDo: check this
+                              )
+        await remove_kbd(bot, message)
```
Comment (Tishka17) [inline aiogram_dialog/manager/manager.py:63]: indent?

File: aiogram_dialog/manager/manager_middleware.py [hunk 5]
```diff
@@ -13,21 +16,71 @@ def __init__(self, registry: DialogRegistryProto):
...
+            await manager.close_manager()
 
-    async def on_post_process_message(self, _, result, data: dict):
-        manager: DialogManager = data.pop("dialog_manager")
-        await manager.close_manager()
 
-    on_post_process_callback_query = on_post_process_message
-    on_post_process_aiogd_update = on_post_process_message
-    on_post_process_my_chat_member = on_post_process_message
+def detect_content_type(update):
```
Comment (Tishka17) [inline aiogram_dialog/manager/manager_middleware.py:50]: 0_o

can we simplify it?

File: aiogram_dialog/context/intent_filter.py (hunk 1 above)
Comment (darksidecat) [reply aiogram_dialog/context/intent_filter.py:65]: This is outer (pre_update) middleware, 
we just skip update,
so maybe just use other Exception for this case?

File: aiogram_dialog/manager/manager.py (hunk 4 above)
Comment (darksidecat) [reply aiogram_dialog/manager/manager.py:63]: pydantic warning, maybe create new class for these cases?

```
pydantic.error_wrappers.ValidationError: 1 validation error for Message
date
  field required (type=value_error.missing)
```

File: aiogram_dialog/context/intent_filter.py (hunk 3 above)
Comment (darksidecat) [reply aiogram_dialog/context/intent_filter.py:147]: oh, missed that.

How to do it better?
Redo the handling of callback.data in filters and ignore prefix?

File: aiogram_dialog/manager/manager_middleware.py (hunk 5 above)
Comment (darksidecat) [reply aiogram_dialog/manager/manager_middleware.py:50]: I was thinking about add to Update property that return event,
or iterate fields that isinstanse(field,  TelegramObject) and get first not None field

File: aiogram_dialog/context/intent_filter.py (hunk 2 above)
Comment (darksidecat) [reply aiogram_dialog/context/intent_filter.py]: fixed

File: aiogram_dialog/manager/registry.py [hunk 6]
```diff
@@ -44,27 +80,36 @@ def register(self, dialog: ManagedDialogProto, *args, **kwargs):
...
-            self.update_handler, *custom_filters, **kwargs
-        )
+    def register_update_handler(self, callback, *custom_filters, run_task=None,
+                                **kwargs) -> None:  # ToDo run_task
+        filters_set = self.update_handler.resolve_filters(kwargs)
+        filters_set.append(*custom_filters)
+
         self.update_handler.register(
-            self.dp._wrap_async_task(callback, run_task), filters_set
+            callback, *filters_set
```
Comment (darksidecat) [inline aiogram_dialog/manager/registry.py:112]: I'm not sure if everything is fine here

**Pull request merged**